salus-config.yaml

```diff
@@ -0,0 +1,5 @@
+scanner_configs:
+  Gosec:
+    exclude:
+      # Deepcopy code generated by OperatorSDK (zz_generated.deepcopy.go)
+      - G601
```

Comment:
We probably don't want to disable this in the entire project, just rather just the one specific line

Reply:
I read over the autogenerated source code and I agree with @nmanoogian that it looks safe -- the loop reference variable is dereferenced and copied to the output.

However I also agree: please let's avoid disabling this for the entire project! It looks like the Salus config for Gosec has the ability to exclude only a directory: https://github.com/coinbase/salus/blob/9d85d3ecbd93f28ed5940ae64bc033df5191cb19/lib/salus/scanners/gosec.rb#L192

Try setting `exclude-dir` in the Salus config for Gosec?

Reply:
I agree -- I'm not a big fan of ignoring G601 for the entire project. Good find, @zbuc! The [docs](https://github.com/coinbase/salus/blob/master/docs/scanners/gosec.md) seem to imply that files are ignored but the comments in the implementation make it seem like files are handled intentionally. I'll make the change to verify it works 👍

Reply:
Eh, it looks like the actual [gosec implementation](https://github.com/securego/gosec/blob/27a5ffb5c8f6dd3b6dea3b8e6019a2b3d43bf0f9/helpers.go#L407) does ignore files after all. I think this leaves these options:

- ❌ Ignore G601 for entire project
- Ignore `api/v1alpha1` (which contains models and simple helpers)
- Add a step to the Makefile which injects G601 ignores into the generated helpers

What do we think?

Reply:
Ah nuts, the actual right way to do this is by using [gosec's `-exclude-generated` flag](https://github.com/securego/gosec#excluding-generated-files). It looks for a specific comment matching regex to pick up patterns like this: `// Code generated by some generator DO NOT EDIT.`. It looks like our SDK conforms to this spec as well. However, Salus doesn't provide a mechanism to pass along the `-exclude-generated` flag to gosec.

Reply:
Lame. I can tell you from experience that the Salus team is open to PRs but can take a while to merge them. If you want, you could feel free to modify and use our fork of Salus in the meantime and submit a PR upstream.

Reply:
I would also be more open to 

> Ignore api/v1alpha1 (which contains models and simple helpers)

Over ignoring G601 project-wide.

Reply:
> I am slightly concerned because the pattern G601 detects is a pretty easy mistake to make and can lead to really subtle and hard to debug failures.

Agreed, I think ignoring `api/v1alpha` is looking like the best option to me right now. Unfortunately though, that is ignoring *all* errors in that directory, not just G601.

Reply:
Is there any way we could have a directory solely containing the autogen code?

Reply:
Unfortunately, OperatorSDK doesn't seem very flexible there.

How much do we hate this last commit where I inject `/* nosec G601 */` into any generated files before Salus runs? 😅

Reply:
Offline: This seems like the lesser of several evils 👍